trace/kafka/option.go

```diff
@@ -61,3 +62,18 @@ func RequiredAcksPolicy(ack RequiredAcks) OptionFunc {
 		return nil
 	}
 }
+
+// Encoder option for injecting a specific encoder implementation
+func Encoder(enc encoding.EncodeFunc, contentType string) OptionFunc {
+	return func(ap *AsyncProducer) error {
+		if enc == nil {
+			return errors.New("encoder is nil")
+		}
+		if contentType == "" {
```

Comment:
why is content type mandatory ?
If someone injects a custom encoder, will they choose a custom content type string ?
Would it make sense to leave the empty string as an option ?
(e.g. the producer does not always look at the content type)

Reply:
The idea is that we should produce a message that has all the bells and whistles... so content type would make the message more complete for a consumer that looke at the content type in order to decode... think about a topic that has mixed encoding...